Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -155,19 +167,31 @@ public IList<ICommandLineArgument> SupportedCommands
     /// The root parameter is used to list the root folder, as the pCloud API
     /// when using oauth tokens creates an isolated folder Applications/ApplicationName
     ///
-    /// When root = true, this is the directory it will be listed.
-    /// When root = false, it will list the _Path directory if it exists, which is
-    /// where the Duplicati backup files are stored.
     /// </summary>
-    /// <param name="root">True to list absolute root, false to list _Path directory</param>
+    /// <param name="folderId">The folder ID to consider as root</param>
     /// <returns></returns>
-    private IEnumerable<IFileEntry> List(bool root)
+    private IEnumerable<IFileEntry> List(ulong folderId)
+    {
+        return ListWithMetadata(folderId)
+            .Select<pCloudFolderContent, IFileEntry>(item => new pCloudFileEntry
+            {
+                IsFolder = item.isfolder,
+                Name = item.name,
+                Size = item.size ?? 0,
+                LastAccess = DateTime.Parse(item.created),
+                LastModification = DateTime.Parse(item.modified)
+            })
+            .ToList();
+    }
+    
+    /// <summary>
+    /// Lists folders with pCloud metadata, necessary to obtain the folder IDs
+    /// </summary>
+    /// <param name="folderId">FolderId to be used as root.</param>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    private IEnumerable<pCloudFolderContent> ListWithMetadata(ulong folderId)
```

Comment:
Consider if this should be `async` and return `Task<List<pCloudFolderContent>>` so we can pass in a cancellation token.

Reply:
Latest commit has implemented this alongside with all cancellation token propagation.